src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,243 @@
+                         .prepareStatement(queryCreateCarAndManufacturer)) {
+            stmtCarManufacturer.setLong(1, id);
+            ResultSet resultSetCar = stmtCarManufacturer.executeQuery();
+            if (resultSetCar.next()) {
+                car = createCar(resultSetCar);
+                Manufacturer manufacturer = createManufacturer(resultSetCar);
+                car.setManufacturer(manufacturer);
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't get the data by id: " + id, exception);
+        }
+        if (car != null) {
+            List<Driver> drivers = getDriversByCarId(car.getId());
+            for (Driver driver: drivers) {
+                car.setDrivers(driver);
+            }
+        }
+        return Optional.ofNullable(car);
+    }
+
+    private List<Driver> getDriversByCarId(Long id) {
+        String queryCreateDriver = "select d.id as driver_id, "
+                + "d.name as driver_name, d.license_number as driver_ln from drivers d "
+                + "inner join cars_drivers cd on cd.driver_id = d.id "
+                + "where cd.car_id = ? and d.deleted = false;";
+
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement stSelectDrivers = connection
+                        .prepareStatement(queryCreateDriver)) {
+            stSelectDrivers.setLong(1, id);
+            ResultSet resultSet = stSelectDrivers.executeQuery();
+            while (resultSet.next()) {
+                Driver driver = createDriver(resultSet);
+                drivers.add(driver);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get drivers by id: " + id, e);
+        }
+        return drivers;
     }
 
     @Override
     public List<Car> getAll() {
-        return Storage.getCars();
+        String query = "SELECT c.id, c.model, c.manufacturer_id, m.name, m.country "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m ON m.id=c.manufacturer_id "
+                + "WHERE c.deleted=FALSE";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            return getAllCarsFromResultSet(resultSet);
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't get all data from DBs", exception);
+        }
     }
 
     @Override
     public Car update(Car car) {
-        List<Car> dbCar = Storage.getCars();
-        IntStream.range(0, dbCar.size())
-                .filter(i -> dbCar.get(i).equals(car))
-                .findFirst()
-                .ifPresent(i -> dbCar.set(i, car));
+        String query = "UPDATE cars c set c.model = ? where c.id = ?;";
```

Comment:
set manufacturer_id as well, and check if car is not deleted

I left this comment earlier as well(